### PR TITLE
refactor: Move GCR logic from service-acccount to GCR module

### PR DESCRIPTION
**Pull request by rmoles**
fixes https://lacework.atlassian.net/browse/ALLY-416
Signed-off-by: Ross Moles <ross.moles@lacework.net>

### CODE REVIEW

File: examples/configure-lacework-gcr-integration/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+```hcl
+provider "google" {}
+
+provider "lacework" {}
+
+module "lacework_svc_account" {
+  source  = "lacework/config/gcp"
+  version = "~> 0.1.1"
+  
+  lacework_integration_name = "Example ECR integration"
```
Comment (afiune) [inline examples/configure-lacework-gcr-integration/README.md:28]: 😅
```suggestion
  lacework_integration_name = "Example GCR integration"
```

File: examples/configure-lacework-gcr-integration/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Create a `main.tf` with the following code:
+
+```hcl
+provider "google" {}
+
+provider "lacework" {}
+
+module "lacework_svc_account" {
+  source  = "lacework/config/gcp"
```
Comment (afiune) [inline examples/configure-lacework-gcr-integration/README.md:25]: ??
```suggestion
  source  = "lacework/gcr/gcp"
```

File: README.md [hunk 3]
```diff
@@ -7,12 +7,39 @@
 
 A Terraform Module to integrate Google Container Registries (GCR) with Lacework.
 
+_**Note:** When `use_existing_service_account` is set to true, the required roles must be added to the Service Account manually._
```
Comment (afiune) [inline README.md:10]: I am hesitant to change the behavior of all the GCP modules (Config, Audit Log, and GCR),
if we are expecting the users to provide the service account and add the roles manually,
what is the sense of adding this option? If we need to go this route, let us just deprecate it
and have the user use the native Terraform resources for Lacework.

My $0.02 👆🏽

Comment (rmoles) [reply README.md]: Agreed. Kudos to @declanwilson-lw. He pointed me in the direction of using the google_iam_policy data provider to check if the role is already applied when using use_existing_service_account=true

Comment (rmoles) [reply README.md]: After further testing on this, I'm not sure that this is working as expected. :(

File: README.md [hunk 4]
```diff
@@ -7,12 +7,46 @@
...
 A Terraform Module to integrate Google Container Registries (GCR) with Lacework.
 
+NOTE: When using an existing Service Account, Terraform cannot work out whether a role has already been applied.
+This mean when running the destroy step, existing roles may be removed from the Service Account. If this Service Account
+is managed by  another Terraform module, you can re-run apply on the other module and this will re-add the role. 
+
+Alternatively, it is possible to remove the offending roles from the state file before destroy, preventing the role(s) 
+from being removed. 
+
+e.g. `terraform state rm 'google_project_iam_binding.for_lacework_service_account'`
```
Comment (afiune) [inline README.md:17]: should we add this command example to all other modules? 🤔

File: examples/configure-lacework-gcr-integration/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+Create a `main.tf` with the following code:
+
+```hcl
+provider "google" {}
+
+provider "lacework" {}
+
+module "lacework_svc_account" {
+  source  = "lacework/gcr/gcp"
+  version = "~> 0.1"
```
Comment (afiune) [inline examples/configure-lacework-gcr-integration/README.md:26]: I am wondering if we should just start with version `1.0.0` like the rest of the modules ... 🤔 

Just a thought 😂

File: main.tf [hunk 6]
```diff
@@ -1 +1,87 @@
...
+    var.service_account_name
+    ) : (
+    length(var.service_account_name) > 0 ? var.service_account_name : "${var.prefix}-lw-${random_id.uniq.hex}"
+  )
+  service_account_json_key = jsondecode(var.use_existing_service_account ? (
+    base64decode(var.service_account_private_key)
+    ) : (
+    base64decode(module.lacework_gcr_svc_account.private_key)
+  ))
+  required_gcr_apis = {
```
Comment (afiune) [inline main.tf:14]: can we move this to be a variable instead of a `local` variable? just like the rest of the modules... 🤔

File: main.tf [hunk 7]
```diff
@@ -1 +1,87 @@
...
 
+resource "random_id" "uniq" {
+  byte_length = 4
+}
+
+data "google_project" "selected" {
+  project_id = var.project_id
+}
+
+data "google_iam_policy" "existing_policy" {
```
Comment (afiune) [inline main.tf:29]: what do we use this for? 🤔

**Review by afiune — APPROVED** [review]
tenor-224724514

File: examples/configure-lacework-gcr-integration/README.md (hunk 5 above)
Comment (rmoles) [reply examples/configure-lacework-gcr-integration/README.md:26]: 🤷 I really don't mind. Happy to update to 1.0.0 if you think that's best

**Pull request merged**